Lab_1/app/src/paid/java/by/bsuir/wildboom/lab1/UnitFragment.java

```diff
@@ -0,0 +1,162 @@
+package by.bsuir.wildboom.lab1;
+
+import android.R.layout;
+import android.os.Bundle;
+import android.text.InputType;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ImageButton;
+import android.widget.Spinner;
+import android.widget.Toast;
+import androidx.fragment.app.Fragment;
+
+import android.content.ClipData;
+import android.content.ClipboardManager;
+import android.content.Context;
+
+import androidx.lifecycle.ViewModelProvider;
+
+public class UnitFragment extends Fragment {
```

Comment:
Remove code duplication by extending UnitFragment from the free version here.

Reply:
I removed code duplication by adding UnitFragmentBase abstract class in main version. In free and paid versions each UnitFragment extends UnitFragmentBase. I did this because I didn't know how a class from the paid version could extend a class from the free version.